crates/oidc-client/src/requests/account_management.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2024 Kévin Commaille.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Methods related to the account management URL.
+//!
+//! This is a Matrix extension introduced in [MSC2965](https://github.com/matrix-org/matrix-spec-proposals/pull/2965).
+
+use serde::Serialize;
+use serde_with::skip_serializing_none;
+use url::Url;
+
+/// An account management action that a user can take, including a device ID for
+/// the actions that support it.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
+#[serde(tag = "action")]
+#[non_exhaustive]
+pub enum AccountManagementActionFull {
```

Comment:
There is some duplication with this struct but that's probably fine for now
https://github.com/matrix-org/matrix-authentication-service/blob/2e8d4da80025c08779b84d971370442dde735b53/crates/router/src/endpoints.rs#L455-L481